.go-version

```diff
@@ -1 +1 @@
-1.23.0
+1.24.6
```

Comment:
Why not use the upstream's 1.25? CI will not be broken because of this ([test PR](https://github.com/openshift/coredns/pull/156)), on local we will have to use 1.25 like upstream maintainers.

Reply:
>  The reason this PR still pins .go-version to 1.24.6 is that we’re also keeping the downstream builder/Dockerfile plumbing aligned with what ART actually ships today, our CI jobs all run inside the golang-1.24 image,

The ART will not use the go version from `.go-version` file because [the coredns image is built from Dockerfile.ocp](https://github.com/openshift/release/blob/5877d5e3a8f05e0a4c86605720accb0910babe2a/ci-operator/config/openshift/coredns/openshift-coredns-main.yaml#L18) which doesn't use the Makefile `make coredns`.

> and local devs can use (and I have been using) the same environment.

That's a valid point. Let's keep the override you put and drop it later when we'll have ART ready for newer go versions.